### PR TITLE
Handle no Athena workgroups

### PR DESCRIPTION
**Pull request by fiskus**
Screenshot from 2021-05-27 11-53-21
Screenshot from 2021-05-27 11-54-05

### CODE REVIEW

File: catalog/app/containers/Bucket/Queries/Athena/Athena.tsx [hunk 1]
```diff
@@ -341,61 +371,65 @@ function State({ children, queryExecutionId }: StateProps) {
...
-                        }
-                      </QueryRunner>
-                    )}
-                  </QueriesFetcher>
-                )
-              }}
-            </QueryResultsFetcher>
-          ),
+          _: ({ value: workgroups }) =>
+            AsyncResult.Init.is(workgroups) || workgroups?.list?.length ? (
```
Comment (nl0) [inline catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: the first argument passed to this function is `AsyncResult`, so `workgroups` cant be `AsyncResult`, bc its a `value` prop of that `AsyncResult`.
something like this should work:
```
 _: (x) =>
  AsyncResult.Init.is(x) || x.value?.list?.length ? (
```

**Review by nl0 — APPROVED** [review]
looking good mostly

File: catalog/app/containers/Bucket/Queries/Athena/Athena.tsx (hunk 1 above)
Comment (fiskus) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: `AsyncResult.Init.is(workgroups)` returns true right now

Comment (fiskus) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: Screenshot from 2021-05-27 17-17-25
Screenshot from 2021-05-27 17-17-36

Comment (nl0) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: 🤔 that's strange, but you can merge if you're confident that it works as expected, tho i'd suggest adding a note to possibly come back to that later

Comment (nl0) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: probably it's bc when result is pending it has previous state as a value

Comment (fiskus) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: I suppose this is more correct:
```tsx
 _: (x) =>
  AsyncResult.Pending.is(x) || x.value?.list?.length ? (
```

Comment (nl0) [reply catalog/app/containers/Bucket/Queries/Athena/Athena.tsx:375]: then probably `AsyncResult.Init.is(x) || AsyncResult.Pending.is(x) || x.value?.list?.length ? (` (if i understand the intent of this condition correctly)

**Pull request merged**